sotodlib/preprocess/processes.py

```diff
@@ -1019,58 +1021,80 @@ class SourceFlags(_Preprocess):
      Example config block::
 
         - name : "source_flags"
-          signal: "signal" # optional
           calc:
             mask: {'shape': 'circle',
-                   'xyr': (0, 0, 1.)}
+                   'xyr': [0, 0, 1.]}
             center_on: 'jupiter'
-            res: 0.005817764173314432 # np.radians(20/60)
-            max_pix: 4e6
+            res: 20 # arcmin
+            max_pix: 4000000
+            distance: 0 # deg
           save: True
           select: True # optional
-    
+
     .. autofunction:: sotodlib.tod_ops.flags.get_source_flags
     """
     name = "source_flags"
-    
+
     def calc_and_save(self, aman, proc_aman):
-        center_on = self.calc_cfgs.get('center_on', 'planet')
-        # Get source from tags
-        if center_on == 'planet':
-            from sotodlib.coords.planets import SOURCE_LIST
-            matches = [x for x in aman.tags if x in SOURCE_LIST]
-            if len(matches) != 0:
-                source = matches[0]
+        center_on_list = self.calc_cfgs.get('center_on', 'planet')
+        source_aman = core.AxisManager(aman.dets, aman.samps)
+        source_list = []
+        for center_on in center_on_list:
+            # Get source from tags
+            if center_on == 'planet':
+                from sotodlib.coords.planets import SOURCE_LIST
+                matches = [x for x in aman.tags if x in SOURCE_LIST]
+                if len(matches) != 0:
+                    source = matches[0]
+                else:
+                    raise ValueError("No tags match source list")
```

Comment:
This doesn't give you a list, it'll always just pick out the first match. I think you want to delete this for loop and to have this check right after `calc_cfgs.get('center_on', 'planet')` then what's called `matches` here can be `source_list` and you don't need `center_on_list` at all you just need `source_list`. Also I don't think `center_on` can ever be planet (L1044) as its a string so when you try to iterate over it it'll fail (addressed via my previous comment) on L1027.